gateway/src/main/java/discord4j/gateway/util/IntentSet.java

```diff
@@ -0,0 +1,279 @@
+     * IntentSet set1 = IntentSet.of(GUILD_MEMBERS);
+     *
+     * set0.or(set1) = IntentSet.of(GUILDS, GUILD_MEMBERS)
+     * }
+     * </pre>
+     *
+     * @param other The other intent set.
+     * @return The union of this set with the other set.
+     */
+    public IntentSet or(IntentSet other) {
+        return IntentSet.of(this.rawValue | other.rawValue);
+    }
+
+    /**
+     * Performs a logical <b>XOR</b> of this intent set with the other intent set.
+     * <p>
+     * The resultant set is the <b>symmetric difference</b> of this set and the other set. A intent is contained if
+     * and only if it was contained in <b>only</b> this set or contained in <b>only</b> the other set.
+     * <pre>
+     * {@code
+     * IntentSet set0 = IntentSet.of(GUILDS, GUILD_MEMBERS, GUILD_BANS);
+     * IntentSet set1 = IntentSet.of(GUILD_BANS, GUILD_EMOJIS);
+     *
+     * set0.xor(set1) = IntentSet.of(GUILDS, GUILD_MEMBERS, GUILD_EMOJIS)
+     * }
+     * </pre>
+     *
+     * @param other The other intent set.
+     * @return The symmetric difference of this set with the other set.
+     */
+    public IntentSet xor(IntentSet other) {
+        return IntentSet.of(this.rawValue ^ other.rawValue);
+    }
+
+    /**
+     * Performs a logical <b>AND NOT</b> of this intent set with the other intent set.
+     * <p>
+     * The resultant set is the <b>relative complement</b> of this set and the other set. A intent is contained if
+     * and only if it was contained in this set and <b>not</b> contained in the other set. This is analogous to
+     * {@link Set#removeAll(java.util.Collection)}.
+     * <pre>
+     * {@code
+     * IntentSet set0 = IntentSet.of(GUILDS, GUILD_MEMBERS, GUILD_BANS);
+     * IntentSet set1 = IntentSet.of(GUILD_MEMBERS, GUILD_BANS, GUILD_EMOJIS);
+     *
+     * set0.andNot(set1) = IntentSet.of(GUILDS)
+     * }
+     * </pre>
+     *
+     * @param other The other intent set.
+     * @return The relative complement of this set with the other set.
+     */
+    public IntentSet andNot(IntentSet other) {
+        return IntentSet.of(this.rawValue & (~other.rawValue));
+    }
+
+    /**
+     * Performs a logical <b>AND NOT</b> of this intent set with the other intent set.
+     * <p>
+     * The resultant set is the <b>relative complement</b> of this set and the other set. A intent is contained if
+     * and only if it was contained in this set and <b>not</b> contained in the other set.
+     * <pre>
+     * {@code
+     * IntentSet set0 = IntentSet.of(GUILDS, GUILD_MEMBERS, GUILD_BANS);
+     * IntentSet set1 = IntentSet.of(GUILD_MEMBERS, GUILD_BANS, GUILD_EMOJIS);
+     *
+     * set0.subtract(set1) = IntentSet.of(GUILDS)
+     * }
+     * </pre>
+     *
+     * @param other The other intent set.
+     * @return The relative complement of this set with the other set.
+     * @deprecated Use {@link IntentSet#andNot(IntentSet)} instead.
+     */
+    @Deprecated
+    public IntentSet subtract(IntentSet other) {
```

Comment:
No need to include this one. We should remove it from PermissionSet in 3.1, too.